test/interpreter_functional/plugins/kbn_tp_run_pipeline/public/components/main.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { unmountComponentAtNode } from 'react-dom';
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+} from '@elastic/eui';
+
+import { RequestAdapter } from 'ui/inspector/adapters/request';
+import { DataAdapter } from 'ui/inspector/adapters/data';
+import { runPipeline } from 'ui/visualize/loader/pipeline_helpers';
+
+// todo: registries must be taken out of package
+import { registries } from '@kbn/interpreter/public';
+
+class Main extends React.Component {
+
+  chartDiv = React.createRef();
+  exprDiv = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    window.runPipeline = async (expression, context = {}, initialContext = {}) => {
+      const adapters = {
+        requests: new RequestAdapter(),
+        data: new DataAdapter(),
+      };
+      console.log('running expression', expression);
+      this.exprDiv.innerText = expression;
+      return await runPipeline(expression, context, {
+        inspectorAdapters: adapters,
+        getInitialContext: () => initialContext,
```

Comment:
Not sure exactly how this works, but why would context ever != initialContext in this case?  I do notice that below you pass in context as empty but initialContext has a time range so maybe there is a reason I'm just not sure I follow it.

Reply:
initialContext is context (initial :P ) that all functions in expression can access. This is how we send in filters and query from dashboard for example.

first context (context being passed to first function) is something completely different. Most of the time we don't use it, but we can for example run the expression with `esaggs` store that output, and provide it as this first context for the expression which starts with `tagcloud` (as we already have the response of esaggs)